scripts/zones/Halvung/npcs/_1qp.lua

```diff
@@ -0,0 +1,37 @@
+-----------------------------------
+-- Area: Halvung
+--  NPC: Decorative Bronze Gate (_1qp)
+-----------------------------------
+local ID = require("scripts/zones/Halvung/IDs")
+require("scripts/globals/npc_util")
+require("scripts/globals/items")
+require("scripts/globals/status")
+
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHas(trade, xi.items.HALVUNG_SHAKUDO_KEY) and
+        npcUtil.tradeHas(trade, xi.items.HALVUNG_BRONZE_KEY) and
+        npcUtil.tradeHas(trade, xi.items.HALVUNG_BRASS_KEY) then
+        player:confirmTrade()
+        npc:openDoor()
+        player:messageSpecial(ID.text.KEY_BREAKS, 2221,2222,2223)
+    end
+end
+
+entity.onTrigger = function(player, npc)
+    if player:getZPos() <= 79 and npc:getAnimation() == xi.anim.CLOSE_DOOR then --from inside the door
```

Comment:
comments are started by `-- ` not `--` (the space) fir legibility reasons in lua, and for execution failure reasons in sql.

Reply:
Also going to want this really well tested before settling on just a Z coordinate check. Not sure what best practice here should be but feels like we should have something more.

Reply:
I too, was curious about this as well on how to check if inside or outside, a lot of other doors that came to mind were banishing gate and Den of Rancor switch. Both of which have Z checks.

Reply:
yeah figuring this out is a work in progress for us, we've been trying to reduce that kind of coordinate checking. Sometimes its unavoidable.

Reply:
how should i proceed with this then? KO offered some advice below: will try that out here in a little bit and give some feedback.

Reply:
be very thorough on testing both sides of that game, maybe also include other axis's. and give some time for others to chime in with any ideas.

Reply:
ok, i think from what I have done so far the Z will suffice because how the area is setup... I will record and post from opposite side and inside; will provide link later.

Reply:
I think what will make this specifically work is how thick the door is but as I said more will be done. Thanks

Reply:
So after a lot of testing and !pos myself I feel confident that if player:getZPos() <= 79.75 is a solid place, which you can only get to by doing !pos. From the locked side you cannot get any closer than 80.8981